### DISCUSSION TITLE
[Project settings][Environments] - I can add a non-production environment inside a production cluster

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#60] [General]
Hi guys, discussion for you.
I noticed that, during the creation of a new environment inside a project, the user can create a `non production` environment inside a cluster used for production and vice versa.

Is this behavior known and wanted?

I think that in the future this should be handled better, is dangerous to allow to users to create new environments that could impact the production cluster and, consequently, the business continuity of a client.

**Comment by fredmaggiowski** [comment]
As of now there's no concept of "production cluster", there are just clusters and it's up to you using them in the appropriate manner; however I think this may be a nice feature to have :)

**Reply by davidebianchi** [reply]
Yes, in addition one cluster could contain both production and non production environments.
So, this is interesting for the future, but more as a rule of the types of environment which could be created (for example, in production cluster it is possible to create only the environment production set in default template of the company)

**Reply by fredmaggiowski** [reply]
exactly, this is how I imagine the feature to be like, also this feature should be optional since some companies may desire to deal with a single cluster while others may want different setups

**Reply by frengooo** [reply]
I completely agree with you. I just wanted to share with you my opinion about that 😄 and reach the discussions record 😂 thanks guys
